samza-api/src/main/java/org/apache/samza/container/placement/ContainerPlacementMessage.java

```diff
@@ -0,0 +1,152 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.container.placement;
+
+import com.google.common.base.Preconditions;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * Encapsulates the request or response payload information between the ContainerPlacementService and external
+ * controllers issuing placement actions
+ */
+public abstract class ContainerPlacementMessage {
+
+  public enum StatusCode {
+    /**
+     * Indicates that the container placement action is created
+     */
+    CREATED,
+
+    /**
+     * Indicates that the container placement action was rejected because request was deemed invalid
+     */
+    BAD_REQUEST,
+
+    /**
+     * Indicates that the container placement action is accepted and waiting to be processed
+     */
+    ACCEPTED,
+
+    /**
+     * Indicates that the container placement action is in progress
+     */
+    IN_PROGRESS,
+
+    /**
+     * Indicates that the container placement action is in progress
+     */
+    SUCCEEDED,
+
+    /**
+     * Indicates that the container placement action is in failed
+     */
+    FAILED;
+  }
+
+  /**
+   * Unique identifier of a request or response message. Helps in identifying duplicate request messages.
+   */
+  protected final UUID uuid;
+  /**
+   * Unique identifier for a deployment so messages can be invalidated across a job restarts
+   * for ex yarn bases cluster manager should set this to app attempt id
+   */
+  protected final String deploymentId;
+  // Logical container Id 0, 1, 2
+  protected final String processorId;
+  // Destination host where container is desired to be moved
+  protected final String destinationHost;
+  // Optional request expiry which acts as a timeout for any resource request to cluster resource manager
+  protected final Duration requestExpiry;
+  // Status of the current request
+  protected final StatusCode statusCode;
+  // Timestamp of the request or response message
+  protected final long timestamp;
+
+  protected ContainerPlacementMessage(UUID uuid, String deploymentId, String processorId, String destinationHost,
+      Duration requestExpiry, StatusCode statusCode, long timestamp) {
+    Preconditions.checkNotNull(uuid, "uuid cannot be null");
+    Preconditions.checkNotNull(deploymentId, "deploymentId cannot be null");
+    Preconditions.checkNotNull(processorId, "processorId cannot be null");
+    Preconditions.checkNotNull(destinationHost, "destinationHost cannot be null");
+    Preconditions.checkNotNull(statusCode, "statusCode cannot be null");
+    Preconditions.checkNotNull(timestamp, "Timestamp of the message cannot be null");
+    this.uuid = uuid;
```

Comment:
We can do null check and assignment in one line:

this.uuid = Preconditions.checkNotNull(uuid, "uuid cannot be null");

same for others.

Reply:
We can use Preconditions.checkNotNull but in one line, e.g.

this.uuid = Preconditions.checkNotNull(uuid, "uuid cannot be null");

Reply:
I misread this, sure changing it